chainerio/filesystems/hdfs.py

```diff
@@ -89,14 +94,39 @@ def info(self):
         info_str = "using hdfs"
         return info_str
 
-    def list(self, path_or_prefix: str = None):
-        if not path_or_prefix or None is path_or_prefix:
+    def list(self, path_or_prefix: str = None, recursive=False):
+        if path_or_prefix is None:
             path_or_prefix = "/user/{}".format(self.username)
 
         self._create_connection()
-        dir_list = self.connection.ls(path_or_prefix)
-        for _dir in dir_list:
-            yield os.path.basename(_dir)
+        target_dir = self.connection.info(path_or_prefix)
+        if "directory" != target_dir['kind']:
+            return None
+
+        target_path = target_dir['path'] + "/"
+        if not path_or_prefix.endswith("/"):
+            path_or_prefix = path_or_prefix + "/"
```

Comment:
Stripping trailing `/` at the beginning of this method is a bit smarter than appending itself IMO, as it would be clearer.